src/code/z_actor.c

```diff
@@ -457,439 +1231,3485 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+    }
+
+    if (!(actorEntry->id & 0x4000)) {
+        rotX *= 0x10000 / 360.0f;
+    } else if (rotX > 180) {
+        rotX -= 360;
+    }
+
+    if (!(actorEntry->id & 0x2000)) {
+        rotZ *= 0x10000 / 360.0f;
+    } else if (rotZ > 180) {
+        rotZ -= 360;
+    }
+
+    return Actor_SpawnAsChildAndCutscene(actorCtx, (GlobalContext*)gameState, actorEntry->id & 0x1FFF,
+                                         actorEntry->pos.x, actorEntry->pos.y, actorEntry->pos.z, rotX, rotY, rotZ,
+                                         actorEntry->params & 0xFFFF, actorEntry->rot.y & 0x7F,
+                                         ((actorEntry->rot.x & 7) << 7) | (actorEntry->rot.z & 0x7F), NULL);
+}
+
+Actor* Actor_Delete(ActorContext* actorCtx, Actor* actor, GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player = GET_PLAYER(globalCtx);
+    Actor* newHead;
+    ActorOverlay* overlayEntry = actor->overlayEntry;
+
+    if ((player != NULL) && (actor == player->unk_730)) {
+        func_80123DA4(player);
+        Camera_ChangeMode(Play_GetCamera(globalCtx, Play_GetActiveCameraIndex(globalCtx)), 0);
+    }
+
+    if (actor == actorCtx->targetContext.arrowPointedActor) {
+        actorCtx->targetContext.arrowPointedActor = NULL;
+    }
+
+    if (actor == actorCtx->targetContext.unk8C) {
+        actorCtx->targetContext.unk8C = NULL;
+    }
+
+    if (actor == actorCtx->targetContext.unk90) {
+        actorCtx->targetContext.unk90 = NULL;
+    }
+
+    func_801A72CC(&actor->projectedPos);
+    Actor_Destroy(actor, globalCtx);
+
+    newHead = Actor_RemoveFromCategory(globalCtx, actorCtx, actor);
+    ZeldaArena_Free(actor);
+
+    if (overlayEntry->vramStart != NULL) {
+        overlayEntry->numLoaded--;
+        Actor_FreeOverlay(overlayEntry);
+    }
+
+    return newHead;
+}
+
+s32 func_800BB59C(GlobalContext* globalCtx, Actor* actor) {
+    s16 x;
+    s16 y;
+
+    Actor_GetScreenPos(globalCtx, actor, &x, &y);
+
+    return (x > -20) && (x < gScreenWidth + 20) && (y > -160) && (y < gScreenHeight + 160);
+}
+
+void func_800BB604(GameState* gameState, ActorContext* actorCtx, Player* player, s32 actorCategory) {
+    GlobalContext* globalCtx = (GlobalContext*)gameState;
+    f32 temp_f0_2;
+    Actor* sp8C;
+    Actor* actor;
+    s32 phi_s2;
+    CollisionPoly* sp80;
+    s32 sp7C;
+    Vec3f sp70;
+    s32 phi_s2_2;
+
+    actor = actorCtx->actorList[actorCategory].first;
+    sp8C = player->unk_730;
+    while (actor != NULL) {
+        if ((actor->update != NULL) && ((Player*)actor != player)) {
+            if (actor->flags & (ACTOR_FLAG_40000000 | ACTOR_FLAG_1)) {
+                if ((actorCategory == ACTORCAT_ENEMY) && CHECK_FLAG_ALL(actor->flags, ACTOR_FLAG_4 | ACTOR_FLAG_1)) {
+                    if ((actor->xyzDistToPlayerSq < SQ(500.0f)) && (actor->xyzDistToPlayerSq < D_801ED8CC)) {
+                        actorCtx->targetContext.unk90 = actor;
+                        D_801ED8CC = actor->xyzDistToPlayerSq;
```

Comment:
```suggestion
                        actorCtx->targetContext.bgmEnemy = actor;
                        sBgmEnemyDistSq = actor->xyzDistToPlayerSq;
```
These are the names recently brought into OoT, they can be applied here to